apollo-router/src/plugins/demand_control/mod.rs

```diff
@@ -306,9 +306,10 @@ impl Plugin for DemandControl {
             let strategy = self.strategy_factory.create();
             ServiceBuilder::new()
                 .checkpoint(move |req: execution::Request| {
-                    req.context
-                        .extensions()
-                        .with_lock(|mut lock| lock.insert(strategy.clone()));
+                    req.context.extensions().with_lock(|mut lock| {
+                        lock.insert(strategy.clone());
+                        lock.insert(req.supergraph_request.body().variables.clone());
```

Comment:
This is going to insert an entry of type `Map<ByteString, Value>` into the context. This type is used both for graphql request variables and graphql response extensions. Should we create some wrapper type to make sure it's disambiguated in context?

Reply:
Yeah, I agree that a wrapper type needs to be used. Maybe also include cost strategy in it so there is just one entry?

I spent a bit of time trying to figure out why we need variables for the response. This is because input fields can have a cost directive right?

Reply:
Yes, we need to include argument cost for a field when scoring responses, and we can't score arguments purely from the schema because we may over/under count optional fields.